pennylane_qiskit/qiskit_device2.py

```diff
@@ -495,36 +448,50 @@ def execute_circuits(session):
     def _execute_sampler(self, circuit, session):
         """Execution for the Sampler primitive"""
 
-        qcirc = circuit_to_qiskit(circuit, self.num_wires, diagonalize=True, measure=True)
-        if circuit.shots:
-            self.options.execution.shots = circuit.shots.total_shots
-        sampler = Sampler(session=session, options=self.options)
+        qcirc = [circuit_to_qiskit(circuit, self.num_wires, diagonalize=True, measure=True)]
+        sampler = Sampler(session=session)
+        compiled_circuits = self.compile_circuits(qcirc)
 
-        result = sampler.run(qcirc).result()
-        self._current_job = result
+        result = sampler.run(compiled_circuits).result()[0]
+        classical_register_name = compiled_circuits[0].cregs[0].name
```

Comment:
Are we sure we only need just one single register? 🤔

Reply:
I'm not 100% sure but it passes all the tests that we have 🤷 

@lillian542 what do you think?

Reply:
Will the order of these classical registers matter, then? Will the measurement registers always remain on the first index?